src/visitor.rs

```diff
@@ -166,39 +166,29 @@ impl<'a, 'v> visit::Visitor<'v> for FmtVisitor<'a> {
 
         match item.node {
             ast::Item_::ItemUse(ref vp) => {
-                match vp.node {
-                    ast::ViewPath_::ViewPathList(ref path, ref path_list) => {
-                        let block_indent = self.block_indent;
-                        let one_line_budget = self.config.max_width - block_indent;
-                        let multi_line_budget = self.config.ideal_width - block_indent;
-                        let formatted = self.rewrite_use_list(block_indent,
-                                                              one_line_budget,
-                                                              multi_line_budget,
-                                                              path,
-                                                              path_list,
-                                                              item.vis,
-                                                              item.span);
-
-                        if let Some(new_str) = formatted {
-                            self.format_missing_with_indent(item.span.lo);
-                            self.changes.push_str_span(item.span, &new_str);
-                        } else {
-                            // Format up to last newline
-                            let span = codemap::mk_sp(self.last_pos, item.span.lo);
-                            let span_end = match self.snippet(span).rfind('\n') {
-                                Some(offset) => self.last_pos + BytePos(offset as u32),
-                                None => item.span.lo
-                            };
-                            self.format_missing(span_end);
-                        }
-
+                let vis = utils::format_visibility(item.vis);
+                let offset = self.block_indent + vis.len() + "use ".len();
+                let context = RewriteContext {
+                    codemap: self.codemap, config: self.config, block_indent: self.block_indent };
+                // 1 = ";"
+                match vp.rewrite(&context, self.config.max_width - offset - 1, offset) {
+                    Some(ref s) if s.len() == 0 => {
```

Comment:
I don't like the idea of using None to mean no result as well as 'can't reformat'. I think it is OK as is - we are requesting rewrite of the view path, not the whole use expression, so it is the caller's responsibility to decide whether to print something or not. It might be cleaner to have this logic outside of the visitor, but I don't think it is urgent.